programs/dynamic-fee-sharing/src/instructions/ix_funding_by_claim_dbc_partner_surplus.rs

```diff
@@ -0,0 +1,116 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token_interface::TokenAccount;
+
+use crate::{
+    error::FeeVaultError,
+    event::EvtFundFee,
+    math::SafeMath,
+    state::{FeeVault, FundingType},
+};
+
+#[event_cpi]
+#[derive(Accounts)]
+pub struct FundingByClaimDbcPartnerSurplusCtx<'info> {
+    #[account(mut)]
+    pub fee_vault: AccountLoader<'info, FeeVault>,
+
+    /// CHECK: The pool config
+    pub config: UncheckedAccount<'info>,
+
+    /// CHECK: The virtual pool
+    #[account(mut)]
+    pub pool: UncheckedAccount<'info>,
+
+    /// The token b account
+    #[account(mut)]
+    pub token_quote_account: Box<InterfaceAccount<'info, TokenAccount>>,
+
+    /// CHECK: The vault token account for quote token
+    #[account(mut)]
+    pub quote_vault: UncheckedAccount<'info>,
+
+    /// CHECK: The mint of token base
+    pub quote_mint: UncheckedAccount<'info>,
+
+    /// CHECK: Token base program
+    pub token_base_program: UncheckedAccount<'info>,
+
+    /// CHECK: Token quote program
+    pub token_quote_program: UncheckedAccount<'info>,
+
+    /// CHECK: dbc pool authority
+    pub dbc_pool_authority: UncheckedAccount<'info>,
+
+    /// CHECK: dbc program
+    #[account(address = dynamic_bonding_curve::ID)]
+    pub dbc_program: UncheckedAccount<'info>,
+    /// CHECK: dbc event authority
+    pub dbc_event_authority: UncheckedAccount<'info>,
+}
+
+pub fn handle_funding_by_claim_dbc_partner_surplus(
+    ctx: Context<FundingByClaimDbcPartnerSurplusCtx>,
+) -> Result<()> {
+    let fee_vault = ctx.accounts.fee_vault.load()?;
+
+    require!(
+        fee_vault
+            .token_vault
+            .eq(&ctx.accounts.token_quote_account.key())
+            && fee_vault.token_mint.eq(&ctx.accounts.quote_mint.key()),
+        FeeVaultError::InvalidFeeVault
+    );
+
+    // support fee vault type is pda account
+    require!(
+        fee_vault.fee_vault_type == 0,
+        FeeVaultError::InvalidFeeVault
+    );
+
+    let before_token_vault_balance = ctx.accounts.token_quote_account.amount;
+
+    let signer_seeds = fee_vault_seeds!(
+        fee_vault.base,
+        fee_vault.token_mint,
+        fee_vault.fee_vault_bump
+    );
+
+    dynamic_bonding_curve::cpi::partner_withdraw_surplus(CpiContext::new_with_signer(
```

Comment:
Could you add a check to see whether pool.is_partner_withdraw_surplus == 0 firstly before we call this, otherwise we just skip the whole flow below?
same for pool.is_creator_withdraw_surplus == 0

Reply:
updated